### PR TITLE
Feature: Blender collect inputs for each instance

### PR DESCRIPTION
**Pull request by Tilix4**
Dependencies aren't fetched correctly because input containers aren't referenced correctly. I've added a collector to link inputs to representations.

## Testing notes:
1. Basically publish (tested on prod_test `E10_SE01_SH001`)
2. Check in loader that the containers in manage are listed as inputs
image

### CODE REVIEW

File: openpype/hosts/blender/api/lib.py [hunk 1]
```diff
@@ -25,6 +26,47 @@
...
 log = Logger.get_logger(__name__)
 
+
+def get_root_containers_from_datablocks(
+    datablocks: List[bpy.types.ID],
+) -> List[bpy.types.ID]:
+    """Get root containers from a list of datablocks.
+    
+    Args:
+        datablocks (List[bpy.types.ID]): Datablocks to get root containers from.
```
Comment (Sharkitty) [inline openpype/hosts/blender/api/lib.py]: Line too long.

File: openpype/hosts/blender/api/lib.py [hunk 2]
```diff
@@ -25,6 +26,47 @@
 
 log = Logger.get_logger(__name__)
 
+
+def get_root_containers_from_datablocks(
+    datablocks: List[bpy.types.ID],
+) -> List[bpy.types.ID]:
+    """Get root containers from a list of datablocks.
+
```
Comment (Sharkitty) [inline openpype/hosts/blender/api/lib.py]: Blank line contains whitespaces.

File: openpype/hosts/blender/api/lib.py [hunk 3]
```diff
@@ -25,6 +26,47 @@
...
+        List[bpy.types.ID]: Root containers.
+    """
+    openpype_containers = bpy.context.window_manager.openpype_containers
+    if not openpype_containers:
+        update_scene_containers()
+
+    # Get all avalon data from openpype_containers as list of dicts
+    all_containers_avalon_data = [
+        c[AVALON_PROPERTY].to_dict()
+        for c in bpy.context.window_manager.openpype_containers
```
Comment (Sharkitty) [inline openpype/hosts/blender/api/lib.py]: Why storing openpype_containers in a variable if you don't use it?

File: openpype/hosts/blender/api/lib.py [hunk 4]
```diff
@@ -25,6 +26,47 @@
...
+            itertools.chain.from_iterable(
+                get_all_outliner_children(d)
+                for d in datablocks
+                if isinstance(d, tuple(BL_OUTLINER_TYPES))
+            )
+        )
+        # and iterate over all provided datablocks
+        | set(datablocks)
+        # Match by the dictionary representation of the avalon data
+        if root_container.get(AVALON_PROPERTY, {})
```
Comment (Sharkitty) [inline openpype/hosts/blender/api/lib.py]: Are you sure you need a fallback to an empty dict? If this check fails it should just stop there and not cause any issue.

**Review by Sharkitty — CHANGES_REQUESTED** [review]
Test passed. Only a few adjustments needed.

File: openpype/hosts/blender/api/lib.py (hunk 3 above)
Comment (Tilix4) [reply openpype/hosts/blender/api/lib.py]: Forgot to refactor that

File: openpype/hosts/blender/api/lib.py (hunk 4 above)
Comment (Tilix4) [reply openpype/hosts/blender/api/lib.py]: Nice catch, it used to be a chained `.get` which became useless with the current implementation

**Review by Sharkitty — APPROVED** [review]

**Pull request merged**